src/server.c

```diff
@@ -5253,6 +5960,39 @@ const char *getSafeInfoString(const char *s, size_t len, char **tmp) {
                        sizeof(unsafe_info_chars)-1);
 }
 
+sds genRedisInfoStringCommandStats(dict *commands) {
+    sds info = sdsempty();
+
+    struct redisCommand *c;
+    dictEntry *de;
+    dictIterator *di;
+    di = dictGetSafeIterator(commands);
+    while((de = dictNext(di)) != NULL) {
+        char *tmpsafe;
+        c = (struct redisCommand *) dictGetVal(de);
+        if (c->calls || c->failed_calls || c->rejected_calls) {
+            sds cmdnamesds = getFullCommandName(c);
+
+            info = sdscatprintf(info,
+                "cmdstat_%s:calls=%lld,usec=%lld,usec_per_call=%.2f"
+                ",rejected_calls=%lld,failed_calls=%lld\r\n",
+                getSafeInfoString(cmdnamesds, sdslen(cmdnamesds), &tmpsafe), c->calls, c->microseconds,
+                (c->calls == 0) ? 0 : ((float)c->microseconds/c->calls),
+                c->rejected_calls, c->failed_calls);
+            if (tmpsafe != NULL) zfree(tmpsafe);
+            sdsfree(cmdnamesds);
+        }
+        if (c->subcommands_dict) {
+            sds subinfo = genRedisInfoStringCommandStats(c->subcommands_dict);
+            info = sdscatsds(info,subinfo);
+            sdsfree(subinfo);
+        }
```

Comment:
maybe it's worth explicitly mentioning in the release notes (top comment) that side effect.
if in the past a call to `CONFIG SET` would appear in cmdstat_config, it will no longer appear there.
so some application that's looking for it there (rather than human eyes) will not find it.